### DISCUSSION TITLE
App Details screen UI improvements

### DISCUSSION THREAD

**Discussion by TheRealMystic** [d4rken-org/permission-pilot#255] [Ideas]
![Screenshot_2022-07-26-11-16-38-66_ce34de6323cfaeb7a2880d42987990b1](https://user-images.githubusercontent.com/54448474/180932620-b6c1b7a2-f7ed-466d-ab08-b22d47bdba82.jpg)

1. Please sort permissions in 3 sections: Allowed (by user) on Top, Denied (by user) at the center, and Managed by System at the bottom.

2. Can you add a filter button (next to or replacing the settings icon on top right) as suggested in #89 ? Default will be **All permissions**, sorted in 3 sections. **Filters must automatically clear/ reset once the user leaves the page.**

3. Black used for Denied permissions isn't too different from the dark blue now in use for Allowed permissions. 

These icons look good in Dark theme where the colours used are different. There is sufficient contrast there to make the allowed permissions stand out.

Can you replace black with something else? Grey would be better than black. 

This also holds true for the screen that shows all apps requesting a given permission, i.e. the screen that opens when user clicks on any permission. Again, please sort the apps alphabetically per section by default:

[List of Apps requesting a given permission](https://user-images.githubusercontent.com/54448474/180937630-4c96a8f2-d5e1-4ffe-a53b-b717af93cd54.jpg)

In this context, please add Sort and Filter options in this screen too (top right):

1. Sort apps alphabetically per section (by default) (i.e. Allowed by user on top, Denied by user at the center, and those managed by System at the bottom)

2. Filters as suggested in #89 .

**Comment by vertigo220** [comment]
The problem with sorting like this is that the allowed permissions will be broken up into multiple sections, making it possible for the user to miss them. Just because the system is allowing an app to use the camera or microphone doesn't mean I don't want to know about it. After all, I don't trust Android/Google to make the best choices for my privacy and security.
